subsys/bluetooth/mesh/pb_adv.c

```diff
@@ -672,12 +641,40 @@ static int bearer_ctl_send_unacked(struct net_buf *buf, void *user_data)
 	prov_clear_tx();
 	k_work_reschedule(&link.prot_timer, PROTOCOL_TIMEOUT);
 
-	bt_mesh_adv_send(buf, &buf_sent_cb, user_data);
+	bt_mesh_adv_send(buf, NULL, user_data);
```

Comment:
This is quite weird. You do not provide callback but provide callback data. Something is actually wrong here.

Since user_data is actively used to pass link statuses or close reason after removing callbacks there is no functionality that applies user_data to the final destination. This change, seems, broke it. The functionality requires much more refactoring or should be reverted. 

This is quite strange why it wasn't caught by bsim tests. Probably we need bsim tests to check provisioning adv bearer behavior.

Reply:
I agree it is wierd. The only purpose the CB's had was to check if the link was closing and to close it if it was. Other than that I could not see another purpose for the CB's. I was not quite sure if I could remove the user_data, so I left them to be discussed here instead.

Reply:
To be honest I just made an attempt to fix this with very little knowledge of the prov layers. It is quite possible that we need a major refactoring to fully resolve this.

Reply:
As far as I understood it: The `buf_sent_cb` was only used from within `bearer_ctl_send_unacked`. All calls to `bearer_ctl_send_unacked` passes a link status as user data. Originally the user data/status was only used when passed to the `close_link` call in the `buf_sent` CB. `buf_start` only purpose was to ensure that `buf_sent` was called if an error occured during sending. When I implemented another scheme for closing the link that respected the `CLOSING_TIMEOUT` define, these CBs did not serve any purpose anymore and could be removed. Am I missing something?